receiver/aerospikereceiver/scraper.go

```diff
@@ -64,86 +64,66 @@ func newAerospikeReceiver(params component.ReceiverCreateSettings, cfg *Config,
 		logger:   params.Logger,
 		config:   cfg,
 		consumer: consumer,
-		clientFactory: func(host string, port int) (aerospike, error) {
-			return newASClient(host, port, cfg.Username, cfg.Password, cfg.Timeout, params.Logger)
+		clientFactory: func(host string, port int) (Aerospike, error) {
+			conf := &clientConfig{
+				host:                  host,
+				port:                  port,
+				username:              cfg.Username,
+				password:              cfg.Password,
+				timeout:               cfg.Timeout,
+				logger:                params.Logger,
+				collectClusterMetrics: cfg.CollectClusterMetrics,
+			}
+			return newASClient(
+				conf,
+				nodeGetterFactory,
+			)
 		},
 		host: host,
 		port: int(port),
 		mb:   metadata.NewMetricsBuilder(cfg.Metrics, params.BuildInfo),
 	}, nil
 }
 
-// scrape scrapes both Node and Namespace metrics from the provided Aerospike node.
-// If CollectClusterMetrics is true, it then scrapes every discovered node
-func (r *aerospikeReceiver) scrape(ctx context.Context) (pmetric.Metrics, error) {
-	errs := &scrapererror.ScrapeErrors{}
-	now := pcommon.NewTimestampFromTime(time.Now().UTC())
-	client, err := r.clientFactory(r.host, r.port)
-	if err != nil {
-		return pmetric.NewMetrics(), fmt.Errorf("failed to connect: %w", err)
-	}
-	defer client.Close()
+func (r *aerospikeReceiver) start(_ context.Context, _ component.Host) error {
+	r.logger.Debug("executing start")
 
-	info, err := client.Info()
+	client, err := r.clientFactory(r.host, r.port)
```

Comment:
I've only had time to take a brief look at this, but one question I immediately have is how this behaves when the client cannot immediately connect.

If I understand correctly, such a scenario will result return an error from the start function. This would cause the entire collector process to exit, which is not desirable. 

Unless there is an unrecoverable problem, we should not return an error here. Additionally, we should be sure that in the case where the client fails to connect during `start`, it will try again later. (Many client do this automatically, but if not, then typically there should be some logic at the beginning of `scrape`.)

Reply:
You are correct, it will return an error.

I think this is what was causing the lifecycle test to fail.

I will look into changing the behavior so this doesn't happen. Maybe I'll try reconnecting in the scrape method and logging an error if the connection fails.